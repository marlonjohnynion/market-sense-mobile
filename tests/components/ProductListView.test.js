import React from 'react'
import configureStore from 'redux-mock-store'
import { shallow } from 'enzyme'
import ConnectedProductsListView from '../../src/containers/ProductListView'

describe('<ProductsListView />', () => {
  const initialState = {
    products: [
      {
        productTitle: 'Carrots',
        productPrice: '₱1500 / 10 kg',
        productLocation: 'OTON, ILOILO',
        productDescription: 'Perfect accompaniment for mains and sandwiches. Ideal edible bed for roasted...',
        productImageUri: 'https://firebasestorage.googleapis.com/v0/b/marketsense-b09f8.appspot.com/o/products%2Fp3.png?alt=media&token=7fd95509-ea2b-4188-b85e-218609d5ee26'
      },
      {
        productTitle: 'Organic Chicken Thighs and Legs',
        productPrice: '₱1000 / 10 kg',
        productLocation: 'LEON, ILOILO',
        productDescription: `Pure organic chicken made in Leon's finest poultry laboratories...`,
        productImageUri: 'https://firebasestorage.googleapis.com/v0/b/marketsense-b09f8.appspot.com/o/products%2Fp4.png?alt=media&token=a26df185-c60c-4a5f-a7d8-4a0d331ccf2a'
      }
    ]
  }

  let container
  let store
  beforeEach(() => {
    store = configureStore()(initialState)
    container = shallow(<ConnectedProductsListView store={store} />)
  })

  it('renders the smart component', () => {
    expect(container).toHaveLength(1)
  })

  it('renders the smart component as expected', () => {
    expect(container).toMatchSnapshot()
  })

  it('receives the props correctly', () => {
    expect(container.prop('products')).toEqual(initialState.products)
  })
})
