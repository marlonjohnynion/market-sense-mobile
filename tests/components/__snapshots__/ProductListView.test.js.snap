// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductsListView /> renders the smart component as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(ProductListView)
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "actions": Object {
        "filterProducts": [Function],
        "searchTermChange": [Function],
        "selectProduct": [Function],
      },
      "filteredProducts": undefined,
      "products": undefined,
      "searchTerm": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "actions": Object {
          "filterProducts": [Function],
          "searchTermChange": [Function],
          "selectProduct": [Function],
        },
        "filteredProducts": undefined,
        "products": undefined,
        "searchTerm": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
